contracts/src/TroveManager.sol

```diff
@@ -663,14 +666,26 @@ contract TroveManager is LiquityBase, ITroveManager, ITroveEvents {
         bool isTroveInBatch = _singleRedemption.batchAddress != address(0);
         uint256 newDebt = _applySingleRedemption(_defaultPool, _singleRedemption, isTroveInBatch);
 
-        // Make Trove unredeemable if it's tiny, in order to prevent griefing future (normal, sequential) redemptions
+        // Make Trove zombie if it's tiny (and it wasn’t already), in order to prevent griefing future (normal, sequential) redemptions
         if (newDebt < MIN_DEBT) {
-            Troves[_singleRedemption.troveId].status = Status.unredeemable;
-            if (isTroveInBatch) {
-                sortedTroves.removeFromBatch(_singleRedemption.troveId);
-            } else {
-                sortedTroves.remove(_singleRedemption.troveId);
+            if (!_singleRedemption.isZombieTrove) {
+                Troves[_singleRedemption.troveId].status = Status.zombie;
+                if (isTroveInBatch) {
+                    sortedTroves.removeFromBatch(_singleRedemption.troveId);
+                } else {
+                    sortedTroves.remove(_singleRedemption.troveId);
+                }
+                // If it’s a partial redemption, let’s store a pointer to it so it’s used first in the next one
+                if (newDebt > 0) {
+                    lastZombieTroveId = _singleRedemption.troveId;
+                }
+            } else if (newDebt == 0) {
```

Comment:
OK, so `newDebt == 0` implies _this_ Trove was fully redeemed, therefore we unzombify it

Reply:
I wouldn’t say “unzombified”. It’s still zombie, but it doesn’t hold the pointer to be the first one in the next redemption sequence anymore.